components/sup/src/manager/spec_watcher.rs

```diff
@@ -12,659 +12,298 @@
     }
+}
 
-    fn run_with<W, P>(path: P) -> Result<Self>
-    where
-        P: Into<PathBuf>,
-        W: Watcher,
-    {
-        let path = path.into();
-        if !path.is_dir() {
-            return Err(sup_error!(Error::SpecWatcherDirNotFound(
-                path.display().to_string()
-            )));
-        }
-        let have_events = Arc::new(AtomicBool::new(false));
-        Self::setup_watcher::<W>(path.clone(), have_events.clone())?;
+impl EnvConfig for SpecWatcherDelay {
+    const ENVVAR: &'static str = "HAB_SPEC_WATCHER_DELAY_MS";
+}
 
-        Ok(SpecWatcher {
-            watch_path: path,
-            have_events: have_events,
-        })
-    }
+// TODO (CM): A strong argument could be made for folding the
+// SpecWatcher functionality into SpecDir itself.
 
-    fn setup_watcher<W>(watch_path: PathBuf, have_events: Arc<AtomicBool>) -> Result<()>
-    where
-        W: Watcher,
-    {
-        thread::Builder::new()
-            .name(format!("spec-watcher-{}", watch_path.display()))
-            .spawn(move || {
-                debug!("SpecWatcher({}) thread starting", watch_path.display());
-                let (tx, rx) = channel();
-                let mut watcher = match W::new(tx, Duration::from_millis(WATCHER_DELAY_MS)) {
-                    Ok(w) => w,
-                    Err(err) => {
-                        outputln!(
-                            "SpecWatcher({}) could not start notifier, ending thread ({})",
-                            watch_path.display(),
-                            err
-                        );
-                        return;
-                    }
-                };
-                if let Err(err) = watcher.watch(&watch_path, RecursiveMode::NonRecursive) {
-                    outputln!(
-                        "SpecWatcher({}) could not start fs watching, ending thread ({})",
-                        watch_path.display(),
-                        err
-                    );
-                    return;
-                }
-                while let Ok(event) = rx.recv() {
-                    debug!(
-                        "SpecWatcher({}) file system event: {:?}",
-                        watch_path.display(),
-                        event
-                    );
-                    have_events.store(true, Ordering::Relaxed);
-                }
-                outputln!(
-                    "SpecWatcher({}) fs watching died, restarting thread",
-                    watch_path.display()
-                );
-                drop(watcher);
-                Self::setup_watcher::<W>(watch_path.clone(), have_events.clone()).unwrap();
-            })?;
-        Ok(())
-    }
+/// Provides an abstraction layer over filesystem notifications for
+/// spec files.
+pub struct SpecWatcher {
+    // Not actually used; only holding onto it for lifetime / Drop
+    // purposes (`Drop` kills the threads that the watcher spawns to do
+    // its work).
+    _watcher: RecommendedWatcher,
```

Comment:
It's not assigning to a field, so much as creating a complete `SpecWatcher` struct (i.e., we never have a complete `SpecWatcher` and then do something like `sw.watcher = my_other_watcher`. 

The underscore prevents the warning that says that nothing else explicitly _uses_ that field.

Reply:
That being said, perhaps slapping `#[allow(dead_code)]` on the `watcher` field is a better way to do this.

Reply:
Hrmm... maybe not. Actually using a field annotated with `#[allow(dead_code)]` does not itself trigger a warning, which is what I think I'd ideally want here. If you as a human find that you're using a field named `_watcher`, though, that ought to trigger some deeper thought on your part.

🤔

Reply:
My comment wasn't to indicate any change desired; I think this is exactly right. I was merely surprised that this:
```rust
        Ok(SpecWatcher {
            _watcher: watcher,
            channel: rx,
        })
```
Doesn't count as _use_, since it's effectively an assignment.